File: 231. Power of Two/main.py
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        val = 1

        while val < n:
            val *= 2

        if val == n:
            return True

        return False


class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n == 1:
            return True
        elif n < 1:
            return False
        elif n % 2 == 0:
            return self.isPowerOfTwo(n / 2)
        else:
            return False


class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n == 1:
            return True

        if n < 1 or n % 2 != 0:
            return False

        return self.isPowerOfTwo(n / 2)


class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        return n > 0 and 1073741824 % n == 0


class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        return n > 0 and (n & (n - 1)) == 0

File: 231. Power of Two/main.rs
struct Solution;

impl Solution {
    pub fn is_power_of_two(n: i32) -> bool {
        let mut val = 1;

        while val < n as i64 {
            val *= 2;
        }

        if val == n as i64 {
            return true;
        }

        false
    }
}

impl Solution {
    pub fn is_power_of_two(n: i32) -> bool {
        if n == 1 {
            true
        } else if n < 1 {
            false
        } else if n % 2 == 0 {
            Self::is_power_of_two(n / 2)
        } else {
            false
        }
    }
}

impl Solution {
    pub fn is_power_of_two(n: i32) -> bool {
        if n == 1 {
            return true;
        }

        if n < 1 || n % 2 != 0 {
            return false;
        }

        Self::is_power_of_two(n / 2)
    }
}

impl Solution {
    pub fn is_power_of_two(n: i32) -> bool {
        n > 0 && 1073741824 % n == 0
    }
}

impl Solution {
    pub fn is_power_of_two(n: i32) -> bool {
        n > 0 && n.count_ones() == 1
    }
}

impl Solution {
    pub fn is_power_of_two(n: i32) -> bool {
        n > 0 && (n & (n - 1)) == 0
    }
}
